pkg/engine/events.go

```diff
@@ -169,7 +169,7 @@ type StepEventStateMetadata struct {
 func makeEventEmitter(events chan<- Event, update UpdateInfo) (eventEmitter, error) {
 	target := update.GetTarget()
 	var secrets []string
-	if target.Config.HasSecureValue() {
+	if target != nil && target.Config.HasSecureValue() {
```

Comment:
Is this change necessary?

Reply:
Yeah—I forget where, but somewhere in the engine code this value is interrogated and it crashes `query`.

Reply:
I thought we used `makeQueryEventEmitter` for query mode?